### DISCUSSION TITLE
Run only one instance of a workflow at any time

### DISCUSSION THREAD

**Discussion by SuperTempAccount** [AutomaApp/automa#1398] [Q&A]
Hey Automars,

Is it possible to limit the number of instances of a workflow that can be triggered at any time?

I have a workflow that is triggered when I visit a website.
I don't want it to run twice if I visit the website again while the first instance of the workflow is still running.

Thanks,

**Comment by Kholid060** [comment]
You can use the workflow state block and set it like the below image

![image](https://github.com/AutomaApp/automa/assets/22908993/099f4d21-9967-43fa-86bd-b8f5761b9a74)

**Comment by SuperTempAccount** [comment]
Hey @Kholid060 are you sure this will work?

I think your suggestion will achieve this behaviour
A is running
B is triggered
B kills A
B is running
C is triggered
C kills B

I want it to work like this
A is running
B is triggered
A kills B
A is running
C is triggered
A kills C

**Comment by flashvnn** [comment]
[Single Workflow.automa.json.zip](https://github.com/AutomaApp/automa/files/12747671/Single.Workflow.automa.json.zip)
You can try this workflow, open https://docs.automa.site/ with multiple tab,  i used localStorage to check workflow is running.

**Reply by Delusionnnn** [reply]
This works if put into my workflow but problem is that if the user does a manual stop command it won't work. Need to be able to run a command if worflow is stopped so that this will work correctly.

**Reply by SuperTempAccount** [reply]
Thanks @flashvnn that's an interesting suggestion. I appreciate the help.
I'll test it out next week to see if I can get it working consistently 🙏

**Comment by SuperTempAccount** [comment]
I tried a `While loop` block to constantly kill other workflows.
It didn't work because the latest workflow can still kill the previous one.

<img width="814" alt="image" src="https://github.com/AutomaApp/automa/assets/83566267/65b4ca08-d3ab-42ec-8a9a-30fbd466a399">
